lib/elixir/lib/integer.ex

```diff
@@ -404,7 +404,7 @@ defmodule Integer do
       0
 
   """
-  @spec gcd(integer, integer) :: pos_integer
+  @spec gcd(integer, integer) :: non_neg_integer
```

Comment:
I think it would be better to add extra clause: `gcd(0, 0) :: 0`.
This way Dialyzer will get more information.